rclcpp/test/rclcpp/test_time_source.cpp

```diff
@@ -212,6 +212,9 @@ TEST_F(TestTimeSource, ROS_time_valid_attach_detach) {
   ts.attachNode(node);
   EXPECT_FALSE(ros_clock->ros_time_is_active());
 
+  EXPECT_FALSE(ros_clock->ros_time_is_active());
+  ts.attachClock(ros_clock);
```

Comment:
clock needs to be added once again here before `detachClock`, because `detachNode` clears node state with associated clocks already.